content/blog/2021-07-13-hyperparam-tuning.md

```diff
@@ -0,0 +1,296 @@
+
+This will produce a table similar to this.
+
+```dvc
+┏━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━┳━━━━━━━━━┳━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━┓
+┃ Experiment              ┃ avg_prec ┃ roc_auc ┃ train.n_est ┃ train.min_split ┃
+┡━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━╇━━━━━━━━━╇━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━┩
+│ workspace               │ 0.51682  │ 0.93819 │ 175         │ 64              │
+│ master                  │ 0.56447  │ 0.94713 │ 100         │ 64              │
+│ └── a1e8716 [exp-09074] │ 0.57333  │ 0.94801 │ 100         │ 32              │
+└─────────────────────────┴──────────┴─────────┴─────────────┴─────────────────┘
+```
+
+### Start tuning with grid search
+
+Now that you've seen how to run an experiment, we're going to write a small
+script to automate grid search for us using DVC. Using grid search in
+hyperparameter tuning means you have an exhaustive list of hyperparameter values
+you want to cycle through. Grid search will cover every combination of those
+hyperparameter values.
+
+We'll do this by creating queues. A queue is how DVC allows us to create
+experiments that won't be run until later. That way we can cycle through
+multiple hyperparameters quickly instead of manually updating a config file with
+new hyperparameter values for each experiment run. The command syntax for
+creating queues looks like this:
+
+```dvc
+$ dvc exp run --queue --set-param train.min_split=8
+```
+
+In the example queue above, we're updating the `train.min_split` value that's
+inside of the `params.yaml` file. This file holds all of the hyperparameter
+values and is where DVC looks to determine if any values have changed. With the
+command above, we're automatically updating that value in the `params.yaml`
+using a queued experiment.
+
+Now we can make the script. You can add a new file to the `src` directory called
+`grid_search.py`. Inside of the file, add the following code.
+
+```python
+import itertools
+import subprocess
+
+# Automated grid search experiments
+n_est_values = [250, 300, 350, 400, 450, 500]
+min_split_values = [8, 16, 32, 64, 128, 256]
+
+# Iterate over all combinations of hyperparameter values.
+for n_est, min_split in itertools.product(n_est_values, min_split_values):
+    # Execute "dvc exp run --queue --set-param train.n_est=<n_est> --set-param train.min_split=<min_split>".
+    subprocess.run(["dvc", "exp", "run", "--queue",
+                    "--set-param", f"train.n_est={n_est}",
+                    "--set-param", f"train.min_split={min_split}"])
+```
+
+This is a simple grid search. We have two hyperparameters we want to tune:
+`n_est` and `min_split`. So we have arrays with a few values in them to mimic
+the exhaustive search a grid search can handle. Then we loop through the values
+and create queued experiments for them using `subprocess`.
+
+You can run this script now and generate your queue with this command.
+
+```python
+$ python src/grid_search.py
```

Comment:
something wrong with this symbol ^^?

Reply:
It was because the codeblock was using `python`. I switched it to use `dvc`.

Reply:
👍